src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -126,6 +132,81 @@ public <E> List<E> listFromTrustedArray(Object[] array) {
                 public <E> List<E> listFromTrustedArrayNullsAllowed(Object[] array) {
                     return ImmutableCollections.listFromTrustedArrayNullsAllowed(array);
                 }
+                @Override
+                public <V> List<StableValue<V>> stableList(int size) {
+                    return StableList.create(size);
+                }
+
+                @Override
+                public <V> V computeIfUnset(List<StableValue<V>> list,
+                                            int index,
+                                            IntFunction<? extends V> mapper) {
+                    if (list instanceof StableList<V> ll) {
+                        return ll.computeIfUnset(index, mapper);
+                    } else {
+                        StableValue<V> stable = list.get(index);
+                        if (stable.isSet()) {
+                            return stable.orThrow();
+                        }
+                        synchronized (stable) {
+                            if (stable.isSet()) {
+                                return stable.orThrow();
+                            }
+                            // Captures
+                            Supplier<V> supplier = new Supplier<>() {
+                                @Override
+                                public V get() {
+                                    return mapper.apply(index);
+                                }
+                            };
+                            return stable.computeIfUnset(supplier);
+                        }
+                    }
+                }
+
+                @SuppressWarnings("unchecked")
+                @Override
+                public <K, V> Map<K, StableValue<V>> stableMap(Set<? extends K> keys) {
+                    K[] arr = (K[]) keys.stream()
+                            .map(Objects::requireNonNull)
+                            .toArray();
+                    return keys instanceof EnumSet
+                            ? StableEnumMap.create(arr)
+                            : StableMap.create(arr);
+                }
+
+
+                @Override
+                public <K, V> V computeIfUnset(Map<K, StableValue<V>> map,
+                                               K key,
+                                               Function<? super K, ? extends V> mapper) {
+                    if (map instanceof HasComputeIfUnset) {
```

Comment:
Can we use pattern matching instanceof here?
```
if (map instance HasComputeIfUnset uc) {
```

Reply:
Good idea.

Reply:
Ahh. I thought you meant pattern matching in another place (which actually turned out to be a really good idea). Here, however, we also need to get the type parameters correct:

<img width="926" alt="image" src="https://github.com/openjdk/jdk/assets/7457876/b2a66033-4c74-42bc-a9de-a976c441ca61">